crates/sui-types/src/multisig.rs

```diff
@@ -185,17 +188,24 @@ impl AuthenticatorTrait for MultiSig {
                 CompressedSignature::ZkLogin(z) => {
                     let authenticator = ZkLoginAuthenticator::from_bytes(&z.0)
                         .map_err(|_| SuiError::InvalidAuthenticator)?;
-                    // Author is already verified against multisig pk, do not verify author within zkLogin authenticator where self.check_author() is evaluated to false for multisig.
+
+                    // Check PublicKey defined in MultisigPublicKey and the one derived from zklogin authenticator is consistent.
+                    if SuiAddress::try_from(&authenticator)? != SuiAddress::from(subsig_pubkey) {
+                        return Err(SuiError::InvalidSignature {
+                            error: "Inconsistent subsig pubkey with authenticator".to_string(),
+                        });
+                    }
+                    // multisig_address is already verified against multisig public key, do not verify multisig_address inside zkLogin authenticator, i.e. self.check_author() is evaluated to false for multisig.
```

Comment:
why do we even pass multisig_address into the zklogin authenticator here? is it used? couldn't we pass in subsig_pubkey, and always verify the author? then we could remove the check_author bool, right?

Reply:
did not think of this before... much better cleaner this way.